config/mock.go

```diff
@@ -84,6 +112,22 @@ func (m *MockConfig) GetRedisHost() (string, error) { return m.GetRedisHostVal,
 func (m *MockConfig) GetMetricsType() (string, error) {
 	return m.GetMetricsTypeVal, m.GetMetricsTypeErr
 }
+func (m *MockConfig) GetHoneycombMetricsConfig() (HoneycombMetricsConfig, error) {
+	hmConfig := &HoneycombMetricsConfig{}
+	err := json.Unmarshal([]byte(m.GetHoneycombMetricsConfigVal), hmConfig)
```

Comment:
do we still need to unmarshal here? We should just allow setting an instance of the struct to the mock?